oauth_device.go

```diff
@@ -32,11 +32,11 @@ func (oa *Flow) DeviceFlow() (*api.AccessToken, error) {
 
 	host := oa.Host
 	if host == nil {
-		host, err := NewGitHubHost("https://" + oa.Hostname)
+		parsedHost, err := NewGitHubHost("https://" + oa.Hostname)
 		if err != nil {
-			return nil, fmt.Errorf("error parsing the hostname '%s': %w", host, err)
+			return nil, fmt.Errorf("error parsing the hostname '%s': %w", oa.Hostname, err)
 		}
-		oa.Host = host
+		host = parsedHost
```

Comment:
shouldn't you also change `os.Host`? that's potentially a difference in the new logic.

Reply:
Good question. No. This was unnecessarily introduced in https://github.com/cli/oauth/pull/63/files#diff-27a48aeb100977fb1eed81f2617982a9a80846825a1be07665a0a8163968d0ceR39

It's never used again. I don't think we should be mutating the struct that the user provided either.